addon/components/nypr-account-forms/login.js

```diff
@@ -42,30 +42,49 @@ export default Component.extend({
     },
     onFailure(e) {
       if (e) {
-        if (get(e, 'errors.code') === 'AccountNotConfirmed') {
-          set(this, 'triedUnconfirmedAccount', true);
-        } else if (get(e, 'errors.code') === 'UnauthorizedAccess') {
-          set(this, 'showLoginError', true);
-        } else {
-          this.applyErrorToChangeset(e.errors, get(this, 'changeset'));
+        let errorCode = get(e, 'errors.code');
+        switch (errorCode) {
+          case 'AccountNotConfirmed':
+            set(this, 'triedUnconfirmedAccount', true);
+            break;
+          case 'UnauthorizedAccess':
+            set(this, 'showLoginError', true);
+            break;
+          default:
+            this.applyErrorToChangeset(e.errors, get(this, 'changeset'));
         }
       }
     },
     loginWithFacebook() {
-      get(this, 'session').authenticate('authenticator:torii', 'facebook-connect')
-      .catch(() => this.onFacebookLoginFailure());
+      let authOptions = {};
+      if (get(this, 'emailWasDeclined')) {
+        authOptions = {authType: 'rerequest'};
```

Comment:
https://developers.facebook.com/docs/facebook-login/permissions/requesting-and-revoking
> If someone has declined a permission for your app, the login dialog won't let your app re-request the permission unless you pass auth_type=rerequest along with your request.

I can include this info as a comment